PubSub/src/PubSubClient.php

```diff
@@ -148,23 +173,39 @@ class PubSubClient
      */
     public function __construct(array $config = [])
     {
-        $this->clientConfig = $config;
-        $connectionType = $this->getConnectionType($config);
         $emulatorHost = getenv('PUBSUB_EMULATOR_HOST');
         $config += [
             'scopes' => [self::FULL_CONTROL_SCOPE],
             'projectIdRequired' => true,
             'hasEmulator' => (bool) $emulatorHost,
-            'emulatorHost' => $emulatorHost
+            'emulatorHost' => $emulatorHost,
+            'transportConfig' => [
+                'grpc' => [
+                    'stubOpts' => ['grpc.max_metadata_size' => 4 * 1024 * 1024,]
```

Comment:
WIth the default metadata size, I was getting a lot of errors saying the metadata has exhausted it's soft limit(specifically for Exactly Once Delivery).

On searching I found, [python](https://github.com/googleapis/python-pubsub/pull/623/files) has also raised this limit to 4MB. So, I added this(so that we only change the default for PubSub).

Reply:
Sounds good, thanks for pointing this out. 
```suggestion
                    // increase default limit to 4MB to prevent metadata exhausted errors
                    'stubOpts' => ['grpc.max_metadata_size' => 4 * 1024 * 1024,]
```